public/app/app.ts

```diff
@@ -293,6 +293,14 @@ function initExtensions() {
   }
 }
 
+function isCrossOriginIframe() {
+  if (window.location.ancestorOrigins.length !== 0 && window.location.ancestorOrigins[0] !== window.location.origin) {
```

Comment:
looks like `ancesterOrigins` isn't supported in firefox (see https://developer.mozilla.org/en-US/docs/Web/API/Location/ancestorOrigins#browser_compatibility)

safer method might be something like:
```tsx
function isCrossOriginFrame() {
  try {
    return document.location.hostname !== window.parent.location.hostname;
  } catch (e) {
    return true;
  }
}
```

[`window.parent` is fully supported](https://developer.mozilla.org/en-US/docs/Web/API/Window/parent#browser_compatibility), but it will throw an error if attempting to access a cross-origin parent's location

Reply:
I really like this approach!